### ISSUE TITLE
HTTPS Everywhere - Redesign & Testing

### ISSUE THREAD

**Issue by elioqoshi** [uracreative/works#29]
## Scope

We will provide Usability and Design services to HTTPS Everywhere, concluding of following:
● User research and survey to understand the current state of the extension better
● Inventory List /Sitemap of all features in the extension
● Design Mockups & Interactive Prototype
● Usability + A/B Testing of current extension compared to new proposed design
● Mockup Revisions based on Usability tests findings
● Accessibility Review and Best practices revision
● Following and support the development process in order to provide assets needed for
a correct implementation

Throughout the process we will be working together with EFF and The Tor Project to ensure
all stakeholders are on the same page. We will set some time aside to meet regularly, do
revisions and address points raised by those parties.

_Note: This work will be focused on the HTTPS Everywhere Extension for Desktop Browsers
(Firefox, Chrome, Opera) and won’t include actual engagement with the Firefox for Android
version of the extension as part of this Scope._

## Action Plan

The planned process will be:
• User research & survey to understand the current shortcomings of the extension
• Mapping out the sitemap and feature inventory
• Wireframes and initial mockups to be revisioned internally
• Interactive prototype of the redesigned extension
• A/B and usability testing of the prototype compared with the extension in production
• Revision of the prototype based on user test results
• Handing it off to the development team at EFF

We expect the redesigned extension to have a big impact on both new and power users,
educating them what HTTPS Everywhere is able to help them with, but also where the
limitations are.

**Comment by elioqoshi** [comment]
@glenn-sorrentino 's original UX audit and design can be tracked here https://github.com/EFForg/https-everywhere/issues/16669

**Comment by elioqoshi** [comment]
@glenn-sorrentino would you be able to provide the design source files you originally used?

**Comment by elioqoshi** [comment]
@zoracon do we have any user research or other details collected from the past? I recall something like that mentioned in our last call.

**Comment by glenn-sorrentino** [comment]
You can find all of my files for this project here: https://keybase.pub/glennsorrentino/HTTPS%20Everywhere/

**Comment by elioqoshi** [comment]
Thanks, that's perfect @glenn-sorrentino

**Comment by elioqoshi** [comment]
Some notes we took during some meetings:
https://pad.ura.design/p/HTTPS-Everywhere-OTF-Usability-Lab

To clarify some points from the notes:

- The Figma prototype won't support any accessibility features like screenreaders or keyboard control. We can only design (to some extent) having [WCAG 2.0 as a guideline](https://en.m.wikipedia.org/wiki/Web_Content_Accessibility_Guidelines). The actual accessibility work needs to be taken into consideration during development, after a report/findings have highlighted its shortcomings.

- We should do some initial lightweight accessibility testing right now of the current extension and note down the findings.

- Please note that many problems might not surface during the testing session with the prototype and we might need to do more testing when a Beta comes out (which hopefully will have resolved potential issues).

**Comment by zoracon** [comment]
Here is a folder with 2 User Interviews and documents of some paper prototyping conducted by Soraya
https://drive.google.com/drive/folders/1E4FJNyAKQoiJeaUfOatBeu74jl2gYeu6?usp=sharing

Let me know if you can't view

**Comment by elioqoshi** [comment]
@zoracon Thanks! The paper prototyping is what Soraya did during Underexposed where we talked and the kickoff for this work happened. The 2 documents are quite useful as well.

I had some help from a friend at Mozilla who is into accessibility. He tested out the current extension quickly with a screenreader and keyboard only. He will share his findings later I hope. From a quick testing session he thought that the extension was pretty accessible already. I'd expect us to keep this in mind when redesigning so we don't break anything which already works. We should wait for a more formal and thorough testing session though.

Also, some additional input I received on Twitter: 
https://twitter.com/stefanjudis/status/1108427561120550913?s=19

**Comment by elioqoshi** [comment]
@zoracon Thanks! The paper prototyping is what Soraya did during Underexposed where we talked and the kickoff for this work happened. The 2 documents are quite useful as well.

I had some help from a colleague at Mozilla who is into accessibility. He tested out the current extension quickly with a screenreader and keyboard only. He will share his findings later I hope. From a quick testing session he thought that the extension was pretty accessible already. I'd expect us to keep this in mind when redesigning so we don't break anything which already works. We should wait for a more formal and thorough testing session though.

Also, some additional input I received on Twitter:
https://twitter.com/stefanjudis/status/1108427561120550913?s=19

**Comment by RenataGegaj** [comment]
This is a draft for the survey we plan to conduct, please leave comments if you have anything to edit, add or remove

How much time do you spend browsing the internet?

- 1-5 hours
- 6-10 hours
- More

Is security and privacy something you are consciously aware of while browsing?

- Yes
- No

What do you know about privacy and security on the web?
How do you tell if a website you are currently using is respecting your privacy? 
When a site requires sensitive information ex: your credit card number, which steps do you take to ensure that that site is credible?

- Look at the sites  FAQ
- Look at the About page
- Read reviews 
- Try to contacts the site developers/maintainers
- Look at the sites URL
- Install app/extension that blocks unsecure sites
- Find out which connection protocol the site uses
- Look at the sites certificate and see if it’s valid
- Use an extension that enforces HTTPS connection on a site
- None
- Other 

Can you tell the difference between HTTP and HTTPS and what do you know about these terms?
What do you know about SSL?
Have you ever used HTTPS Everywhere?

- Yes
- No

How familiar are you with HTTPS Everywhere, please elaborate on what you think it does for you?
Which statements do you associate with HTTPS Everywhere

- Browser extension that blocks insecure sites
- Browser extension that enforces sites to use HTTPS
- Browser extension that blocks the non secure elements of the site
- A protocol that indicates a secure connection

What does HTTPS Everywhere protect you against?

**Comment by elioqoshi** [comment]
@holantonela do you have some feedback for the survey questions?

**Comment by holantonela** [comment]
I think the questions are great. It will give you some idea about how familiar users are with privacy and security online and please, don't feel disappointed if that familiarity is very low. 

Question 5 - Is it a multiple-answer multiple choice type of question? You will need checkbox there.
Question 7 - You may want to have needs options. You will get very random answers there. 
Question 8 - I'd include a third option that covers  _I don't know what it is_. 

Don't you want to know which browser they are using?

**Comment by RenataGegaj** [comment]
@holantonela Great edits, I'll add them right now.
> Question 5 - Is it a multiple-answer multiple choice type of question? You will need checkbox there.

Yes, the questions with bullet points are all multiple choice and we'll replace them with checkboxes on the survey.

**Comment by elioqoshi** [comment]
@RenataGegaj I'd suggest adding a 1-5 point option with "How concerned are you of your browsing always using a HTTPS connection" (might need improvements on the wording.

> What do you know about SSL?

Can we say "What does SSL mean for you?" it might feel less like a school test question.

> Is security/privacy something you are aware of while browsing?

I'm not sure of the wording of this one. Also this should not be about privacy I suppose. Maybe "Is being on a secure connection something you are aware of while browsing?"

> How much time do you spend browsing the internet?

Is this per week? Should be specified.

Looks solid to me otherwise

**Comment by elioqoshi** [comment]
Oh, how about:

"Do you want to join our Beta Testers Team? We might ask for your feedback again once we redesign the HTTPS Everywhere experience"

So we can get back to interested tester's later on without reaching out from scratch?

**Comment by RenataGegaj** [comment]
Thanks for the feedback @elioqoshi I made the last edits and tested the draft in person with a user. Everything went smoothly, the survey is ready to be published

**Comment by elioqoshi** [comment]
Survey here: https://oasis.sandstorm.io/grain/4r6jnb2vkkveFiyC78MjJQ/

**Comment by elioqoshi** [comment]
We were made aware that the Sandstorm QuickSurvey had Accesssibility problems (ironically). We wanted to offer a Google Forms alternative for those who want to avoid using Google Form. However, it seems to be the best option right now while still being accessible. The link to the same survey, but on Google Form can be found here:

https://forms.gle/YZ9FCAA5UnypwvjY7

**Comment by elioqoshi** [comment]
This is going well, we finished the first draft of the main doorhanger UI:
https://www.figma.com/file/Fv6ivAdU7z4MxG0jU5uOduFe/HTTPS-Everywhere-Redesign?node-id=13%3A26

Now we are tackling the UI in the Add-On Settings and hope to start some first user testing next week.

**Comment by elioqoshi** [comment]
@RenataGegaj made a short report of the user survey  (32 responses) and we did a quick analysis of the survey results. These are too raw for publishing yet, but I'd suggest we publish them once it's a bit cleaned up and anonymized.

**Comment by theel0ja** [comment]
The new design doesn't fit to the extension window properly (Firefox 60.6.1esr)

`+` it requires extra clicks to find the Stable rules used in current one (and that doesn't fit to the window either)

![New design](https://user-images.githubusercontent.com/5832930/57127063-bd417000-6d7e-11e9-9718-37dc5fa95363.png)

old one shows it without extra clicks and scrolling:

![Old design](https://user-images.githubusercontent.com/5832930/57127064-be729d00-6d7e-11e9-9ad0-fe92a5ddfe7e.png)

The buttons should be smaller

**Comment by theel0ja** [comment]
For example, macOS wastes much less space for buttons in Settings menu

![macOS Settings](https://s14-eu5.startpage.com/cgi-bin/serveimage?url=https:%2F%2Fwww.imore.com%2Fsites%2Fimore.com%2Ffiles%2Fstyles%2Fxlarge%2Fpublic%2Ffield%2Fimage%2F2016%2F12%2Fgatekeeper-os-x-anywhere-setting.jpg%3Fitok%3DoD6g9MEo&sp=4a45b9228be48634745697a43388eb22)

**Comment by theel0ja** [comment]
Even macOS wastes less space for button sizes

![](https://cdn1.tekrevue.com/wp-content/uploads/2018/09/macos-mojave-settings-dark-mode-light.png)

**Comment by elioqoshi** [comment]
@theel0ja thanks for bringing this up but this issue is not about this latest version which is a minor redesign. I'd suggest you open these issues seperately in the relevant repo. cc @zoracon 

The redesign in this issue is addressing a redesign beyond the changes of the last version.

**Comment by zoracon** [comment]
@elioqoshi Sorry you're getting the overflow of backlash to the UI changes. 
If anyone comes here with issues please file at https://github.com/EFForg/https-everywhere/issues

**Comment by elioqoshi** [comment]
@zoracon no worries, just wanted to clarify as we might not have been clear about it in the issue before as well :)

**Comment by elioqoshi** [comment]
@zoracon are the new temporary maintainers involved in this issue? We should really get started with the new prototype and the new design implemented.

**Comment by elioqoshi** [comment]
There is a really good Firefox UX audit of the protection panels here:
https://mozilla.invisionapp.com/share/38S2U6FPFKD#/screens/359913354

This is a good base to do further iterations on the design.

**Comment by elioqoshi** [comment]
I went ahead and did another iteration based on the work from the Firefox UX team as it's very close to patterns used in HTTPS Everywhere. I propose that we have one more feedback session on this iteration internally and do one more Usability testing session.

Here is the UX walkthrough of the next iteration:

https://www.figma.com/proto/Fv6ivAdU7z4MxG0jU5uOduFe/HTTPS-Everywhere-Redesign?node-id=367%3A378&viewport=35%2C635%2C0.4026089310646057&scaling=min-zoom

Alternatively, here is the exported presentation:
![Presentation](https://user-images.githubusercontent.com/5436686/59966839-242f0a00-9522-11e9-94c2-ecb14ec5d9e4.jpg)

**Comment by RenataGegaj** [comment]
As we are rounding up HTTPS Everywhere redesign, I'm sharing here the data that we gathered from our survey. Listed here you can find a short paper the with conclusions from the results(on the pdf) or if you are interested in exploring the raw data, find that linked below(on the sheet link). 
If you have any comments/questions please comment here, we'd love to discuss further...

I will share the final paper of the whole redesign process very soon, stay tuned for that ;)

https://docs.google.com/spreadsheets/d/1gsbNN9sfoEnyFrE4sr2iQHk9ZldhrWQ9OUWSncYB84g/edit?usp=sharing

[HTTPSE Survey report.pdf](https://github.com/uracreative/works/files/3505022/HTTPSE.Survey.report.pdf)

**Comment by zoracon** [comment]
One, outstanding job on not only utilizing feedback but using a well adjusted design pattern.

Things that really stand out as great improvements:

**Iconography** - The "disabled for this site" icon was a tough challenge and I believe this is a great iteration of it

**States**
Moving states into a more unified fashion helped clear up the ecology system of the extension itself.  

There are many moving parts to extensions. But it has proven to just fragment user understanding over time.

One complaint from advanced users who add custom rules is that they did not like having to "click more" to reach the area to add a rule, but maybe with this presentation, their opinion could be different

Will be sharing this internally. However is feedback from other HTTPS Everywhere contributors and users welcomed as well? I can link this issue for more discussion.

I'm hoping with this new design iteration to have an "what's been updated" splash page. To onboard everyone next time. That was a mistake I made with the refresh in May.

**Comment by elioqoshi** [comment]
@zoracon Thank you for the feedback and the kind words on the redesign!

> Will be sharing this internally. However is feedback from other HTTPS Everywhere contributors and users welcomed as well? I can link this issue for more discussion.

Of course, please do.

> I'm hoping with this new design iteration to have an "what's been updated" splash page. To onboard everyone next time. That was a mistake I made with the refresh in May.

Within the extension or separately on the EFF website? We can help with that if it's the former.

A few more things to note:

- It's updated on Figma but not in the exported image preview: we unified the 2 toggle buttons as that created a lot of confusion during user testing. In this new design we envision that it would not be possible to enable the extension globally but not block unencrypted connections. With the change of the SSL landscape we think this is the right move and simplifies a lot of user decisions. You should discuss this within the development team as well.

- There was some confusion regarding the Rules icon which felt out of place for some users. We haven't found an alternative yet but would like to update that as well.

**Comment by elioqoshi** [comment]
Here is the updated Figma export on the redesign:

![image](https://user-images.githubusercontent.com/5436686/64411700-16ebdb00-d08e-11e9-83ce-3dc0aff4761f.png)

**Comment by zoracon** [comment]
@elioqoshi Apologies for the delay in response. We have feedback internally from our design lead on HTTPSE, Soraya. Her feedback is below. 

Once feedback is consolidated, we can reconnect. I'll be sure to watch this ticket more closely.

>>>
—
Hi Elio, Anxhelo, Renata and broader Ura design team!
I’m sorry it’s taken me a while to get back to you all! It takes me a while to read through things carefully, and wanted to make sure your efforts received the proper attention they deserved. Thank you so much for all your efforts on this project.

Best,
Soraya


A small note on crediting:
A whole bunch of people helped with the project, ranging from Glenn Sorrentino (@glenn-sorrentino)  for kicking off the conversation on a redesign, to myself, to Alexis for project managing, to Georgia, Molly, Ame, and Bernard through Simply Secure during the UnderExposed design week (how I met Anxhelo and Elio), to the user research folks at Tor, to the recruited testers who provided the user feedback. It would be worthwhile and appreciated to acknowledge their contributions (with their consent, of course), as these things are often a large team effort with a lot of invisible work behind the scenes. Relatedly, it would be good to know how specific people on the team at Ura contributed, and to name folks if they’re comfortable (this happens on the acknowledgments page of the PDF, but it might be worth floating this higher, and across the different versions of this report).

Survey:
The survey is fantastic. One thing I might revisit for the future is question 7: “When a site requires sensitive information…what steps do you take to ensure that the site is credible?” To me, this user group is a bit unusual — in our context in the US, I’m a little doubtful that most people have awareness around this, nor around what a man-in-the-middle attack is. If we can repurpose this question in the future, I might consider reframing it so that it’s more open-ended in its answers, so as to account for users who may have less knowledge around this, and to consider framings people may not have organically considered.
I really like the framing for questions 9, 10 and 11.

Personas:
I liked these initial personas, but would have loved to learn more about what was unique for their context in the testing group and to have further details. If they were based on the testers in Albania, it would have been nice to know what a common browser was, what kind of things people are looking at, as we don’t get to hear consolidated user stories like that from our perspective here in San Francisco.

Scenario tasks:
+ I like the initial premise. I think if we were to repurpose/recycle any of the prompts, I might suggest rewording slightly from “make sure you don’t enter sites that don’t use encryption” since it includes two negatives, and instead maybe more directly say “avoid HTTP sites” or an alternate framing that moves closer to what wording is intended. 
+ Interacting with the rules for HTTPS Everywhere is interesting and a use case I don’t often anticipate! I really appreciated that these were two of the discreet tasks, and the chart that you highlighted for the various iterations were interesting. I wonder which users were fairly comfortable with this framing. I’m not sure if appending a new rule is commonly engaged with by new users (referring to questions 4 and 5), but it’s regardless interesting to see whether these features performed okay in usability.

The report itself was interesting—I really liked reading through the iterations and rationale for changes. Thanks for making this open for others to read and enjoy.

In response to specific pages in the report: 
Pg 21 - Great observations on the help icon and “disable for this site” button. On our end: we should reassess how to make these indicators / where they lead clearer.
Pg 22 - I completely agree that clearer indicators are needed for showing the changed state of the current state, as well as prioritizing “disable for this site”, given that users are less likely to need/ interact with the other features.
Pg 23 - I like the proposal for the strike-through grayed out logo with the small dot, indicating the disabled-for-the-current-site-but-enabled-globally use case. 


Proposed redesign:

+ I really like the influence from Mozilla’s Photon system that y’all and Glenn pointed to. Historically, we haven’t been directly influenced by these browser-specific design systems, so it’s a nice reference point as we now have a bit more capacity on our design team to think through brand consistency, standard alignment and cohesion across our projects.

+ I like the phrasing for “allow only secure requests” — we have been playing with wording around this from “encrypt all eligible sites” and “unencrypted sites are allowed”, without coming to a good place just yet. I think we might have to do some user testing in this area to find something that’s intuitive. 

+ I’m a little hesitant to use these terms in particular: “global settings” and “site settings”. I realize they are legacy terms borrowed from the software development space and also historically used on the extension, but I think that we can find something that’s perhaps more intuitive for a new user thrown into the extension for the first time. I think this is something that we’ll probably be discussing more and scratching our heads around, perhaps based on further user testing and research.

+ I wonder what can be drawn from all three design iterations, from Glenn’s, myself and from Ura Design’s. Internally at EFF, we’re trying to find ways to more prominently show that our browser extensions and transport layer encryption projects (e.g. Privacy Badger and Certbot) are projects of EFF, since people don’t always make those associations. One challenge we’ll anticipate is also including that information (+ EFF branding), as well as our HTTPS Everywhere project branding.  Additionally, a priority for us (in addition to making the extension more accessible and usable), is to better facilitate donations for our nonprofit, and to find more ways to point people to our organization’s work.

+ On the internal end, we are having quite a few conversations about how to unify our styling, typography and iconography between our projects, so our design and software development teams likely will be interpreting which pieces to keep from the placement / icons / wording / visual cues /usability, and might be making different decisions for the extension’s look and feel. The user research in particular will be super helpful for us to point to, as well as the rationale between decisions in the different iterations of the extension wireframes.

Hope this feedback is helpful! Thanks again for your work on this.

**Comment by elioqoshi** [comment]
Thanks @zoracon ! We gathered all of the feedback and will address it soon. We will update you by the end of this month regarding a new revision.

In the meantime, there is more feedback from @epang who is a UX Designer at Mozilla:

![image](https://user-images.githubusercontent.com/5436686/66213888-e913a980-e6c0-11e9-9778-7983cc8b3081.png)

**Comment by elioqoshi** [comment]
Closing this and we can open up a new issue when the design is being implemented and the other changes can be addressed. 

From the OTF side this has been wrapped up last year already.
